src/bodypart.cpp

```diff
@@ -561,32 +561,23 @@ void body_part_type::check() const
 
 float body_part_type::get_limb_score( const limb_score_id &id ) const
 {
-    for( const bp_limb_score &bpls : limb_scores ) {
-        if( bpls.id == id ) {
-            return bpls.score;
-        }
+    if( limb_scores.count( id ) ) {
+        return limb_scores.at( id ).score;
     }
     return 0.0f;
```

Comment:
might be worth grabbing iterator instead, considering the recent perf issues - you can cut another search then e.g.
```suggestion
    const auto it = limb_scores.find( id );
    return it == limb_scores.end() ? 0.0f : *it;
```

Reply:
good call. required a minor syntax change, but this is in now.